account_invoice_fiscal_position_update/models/account_invoice.py

```diff
@@ -0,0 +1,73 @@
+# © 2011-2014 Julius Network Solutions SARL <contact@julius.fr>
+# © 2014 Akretion (http://www.akretion.com)
+# @author Mathieu Vatel <mathieu _at_ julius.fr>
+# @author Alexis de Lattre <alexis.delattre@akretion.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+
+from odoo import models, api, _
+
+
+class AccountInvoice(models.Model):
+    _inherit = "account.invoice"
+
+    @api.onchange('partner_id')
+    def _onchange_partner_id(self):
+        fiscal_position = self.fiscal_position_id
+        res = super(AccountInvoice, self)._onchange_partner_id()
+        if fiscal_position != self.fiscal_position_id:
+            self.fiscal_position_change()
+        return res
+
+    @api.onchange('fiscal_position_id')
+    def fiscal_position_change(self):
+        """Updates taxes and accounts on all invoice lines"""
+        self.ensure_one()
+        res = {}
+        lines_without_product = []
+        fp = self.fiscal_position_id
+        inv_type = self.type
+        for line in self.invoice_line_ids:
+            if line.product_id:
+                product = line.product_id
+                if inv_type in ('out_invoice', 'out_refund'):
+                    account = (
+                        product.property_account_income_id or
```

Comment:
You should ensure the company of the property for multicompany environments 
`with_context(force_company=self.company_id.id)`